src/server/services/generation/generation.service.ts

```diff
@@ -337,11 +337,24 @@ const baseModelToOrchestration: Record<BaseModelSetType, string | undefined> = {
   SDXLDistilled: 'SDXL_Distilled',
 };
 
+async function checkGenerationAvailability(resources: CreateGenerationRequestInput['resources']) {
+  const data = await getResourceData(resources.map((x) => x.id));
+
+  return data.every((x) => !!x.generationCoverage?.covered);
+}
+
 export const createGenerationRequest = async ({
   userId,
   resources,
   params: { nsfw, negativePrompt, ...params },
 }: CreateGenerationRequestInput & { userId: number }) => {
+  if (!resources || resources.length === 0) throw throwBadRequestError('No resources provided');
+  if (resources.length > 10) throw throwBadRequestError('Too many resources provided');
+
+  const allResourcesAvailable = await checkGenerationAvailability(resources).catch(() => false);
```

Comment:
I thought we were already filtering out uncovered resources on the frontend. Have we run into any issues with this? Or is this just to make the service more robust?